src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,108 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useContext, useState } from 'react';
+import { PostContext } from './context/PostContext';
+import { Error } from '../types/Error';
+import { Comment } from '../types/Comment';
+import { postComment } from '../api/comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  selectedPost: Post | null,
+};
+
+export const NewCommentForm: React.FC<Props> = React.memo(({
+  selectedPost,
+}) => {
+  const {
+    setError,
+    setComments,
+    isLoading,
+    setIsLoading,
+  } = useContext(PostContext);
+
+  const [authorName, setAuthorName] = useState('');
+  const [isNameError, setIsNameError] = useState(false);
+
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+
+  const [commentText, setCommentText] = useState('');
+  const [isCommentError, setIsCommentError] = useState(false);
+
+  const handleAuthorName = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsNameError(false);
+    setAuthorName(e.target.value);
+  };
+
+  const handleAuthorEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsEmailError(false);
+    setAuthorEmail(e.target.value);
+  };
+
+  const handleCommentText = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setIsCommentError(false);
+    setCommentText(e.target.value);
+  };
+
+  const handleSubmitForm = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+
+    if (!authorName) {
+      setIsNameError(true);
+    }
+
+    if (!authorEmail) {
+      setIsEmailError(true);
+    }
+
+    if (!commentText) {
+      setIsCommentError(true);
+    }
+
+    if (!authorName || !authorEmail || !commentText) {
+      return;
+    }
+
+    let newComment: Omit<Comment, 'id'>;
+
+    try {
+      if (selectedPost) {
+        newComment = {
+          postId: selectedPost?.id,
```

Comment:
There is no need to use optional chaining here because you already checked if 'selectedPost' is not null at line 70.